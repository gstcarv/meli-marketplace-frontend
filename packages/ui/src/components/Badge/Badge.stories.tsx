import { TruckIcon } from '@meli/icons';
import type { Meta, StoryObj } from '@storybook/react';
import { Badge } from './Badge';

const meta = {
    title: 'core/Badge',
    component: Badge,
    parameters: {
        layout: 'centered'
    },
    tags: ['autodocs']
} satisfies Meta<typeof Badge>;

export const Primary: StoryObj<typeof meta> = {
    args: {
        children: (
            <TruckIcon size={11} color='#ffffff' />
        ),
        label: 'Badge example',
        color: 'accent-500'
    }
};

export const Accent: StoryObj<typeof meta> = {
    args: {
        children: (
            <TruckIcon size={11} color='#ffffff' />
        ),
        label: 'Badge example',
        color: 'primary-500'
    }
};

export const Success: StoryObj<typeof meta> = {
    args: {
        children: (
            <TruckIcon size={11} color='#ffffff' />
        ),
        label: 'Badge example',
        color: 'success-500'
    }
};

export default meta;
