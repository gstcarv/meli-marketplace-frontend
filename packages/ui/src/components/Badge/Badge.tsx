import { Colors } from '@meli/design-tokens';
import styled from 'styled-components';

type Props = {
    color?: Colors;
};

export const Badge = styled.div<Props>`
    width: 22px;
    height: 22px;
    border-radius: ${({ theme }) => theme.radii.get('full')};
    background: ${({ color = 'primary-500', theme }) => theme.colors.get(color)};
    display: flex;
    justify-content: center;
    align-items: center;
`;
