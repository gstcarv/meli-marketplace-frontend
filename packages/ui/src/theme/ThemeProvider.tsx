import 'react-loading-skeleton/dist/skeleton.css';
import { PropsWithChildren } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { GlobalStyle } from '../globalStyles';
import { theme } from './theme';

export const ThemeProvider = (props: PropsWithChildren) => (
    <StyledThemeProvider theme={theme}>
        <>
            {props.children}

            <GlobalStyle />
        </>
    </StyledThemeProvider>
);
