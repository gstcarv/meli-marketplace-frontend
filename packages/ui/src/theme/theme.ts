import { colors, fontSize, radii, spacing, fontWeight } from '../tokens';

export const theme = { colors, fontSize, radii, spacing, fontWeight };

export type Theme = typeof theme;
