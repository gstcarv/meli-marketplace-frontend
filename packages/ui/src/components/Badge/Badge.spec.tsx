import { render } from '@testing-library/react';
import { ThemeProvider } from '../../theme';
import { Badge } from './Badge';

describe('<Badge />', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ThemeProvider>
                <Badge>2</Badge>
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});
