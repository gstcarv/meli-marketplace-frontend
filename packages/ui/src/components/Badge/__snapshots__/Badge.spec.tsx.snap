// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render correctly 1`] = `
<div>
  <div
    class="sc-blHHSb jHmzXs"
    role="alert"
  >
    2
  </div>
</div>
`;
